src/utils/contractHelpers.ts

```diff
@@ -31,9 +32,35 @@ export const makeVaultTxFn =
   };
 
 export const makeRouterTxFn =
-  (web3Context: Web3ReactContextInterface, contracts: Contracts) =>
+  (
+    web3Context: Web3ReactContextInterface,
+    contracts: Contracts | undefined,
+    apeError: (error: any) => void
+  ) =>
   async (callback: (apeRouter: ApeRouter) => Promise<ContractTransaction>) => {
+    if (!contracts) {
+      apeError('Contracts not loaded');
+      return;
+    }
     const signer = await web3Context.library.getSigner();
     const apeRouter = contracts.apeRouter.connect(signer);
     return callback(apeRouter).catch(e => handleContractError(e));
   };
+
+export const makeDistributorTxFn =
+  (
+    web3Context: Web3ReactContextInterface,
+    contracts: Contracts | undefined,
+    apeError: (error: any) => void
+  ) =>
+  async (
+    callback: (apeDistributor: ApeDistributor) => Promise<ContractTransaction>
+  ) => {
+    if (!contracts) {
+      apeError('Contracts not loaded');
+      return;
+    }
+    const signer = await web3Context.library.getSigner();
+    const apeDistributor = contracts.apeDistributor.connect(signer);
```

Comment:
@Niraj-Kamdar this is no longer necessary, since I changed `Contracts` to accept a signer, so all the contracts are connected already